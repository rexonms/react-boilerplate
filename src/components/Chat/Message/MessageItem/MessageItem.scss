.container {
  display: flex;
  justify-content: flex-start;
  align-items: center;
  margin-bottom: 10px;
  padding-bottom: 15px;
  border-bottom: 1px solid #eee;
}
.avatar {
  margin-right: 10px;
}

.details {
  p {
    margin: 0;
  }
}