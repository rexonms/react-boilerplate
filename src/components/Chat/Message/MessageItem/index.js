import React, { Component } from 'react';
import PropTypes from 'prop-types';

import styles from './MessageItem.scss';
import Avatar from '../../../Avatar/index';
import Heading6 from '../../../Typography/H6/index';
import Paragraph from '../../../Typography/Paragraph/index';

const propTypes = {
  /** Image that will be displayed as Avatar */
  avatarImageURL: PropTypes.string.isRequired,
  /** Name of the message recipient */
  recipientName: PropTypes.string.isRequired,
  /** The last message that was sent to the recipient */
  lastMessage: PropTypes.string.isRequired,
};

const defaultProps = {};

class MessageItem extends Component {
  constructor (props) {
    super(props)
    this.state = {
    }
    this.onClickHandler = this.onClickHandler.bind(this);
  }
  onClickHandler() {
    alert(`Start chatting with ${this.props.recipientName}`)
  }

  render () {
    return (
      <div className={styles.container} onClick={this.onClickHandler}>
        <div className={styles.avatar}>
          <Avatar imageURL={this.props.avatarImageURL} />
        </div>
        <div className={styles.details}>
          <Heading6 labelText={this.props.recipientName} />
          <Paragraph label={this.props.lastMessage} size='small' />
        </div>
      </div>
    )
  }
};

MessageItem.defaultProps = defaultProps;
MessageItem.propTypes = propTypes;

export default MessageItem;
