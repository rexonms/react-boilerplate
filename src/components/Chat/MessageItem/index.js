import React from 'react';
import PropTypes from 'prop-types';

import styles from './MessageItem.scss';
import Avatar from '../../Avatar/index';
import Paragraph from '../../Typography/Paragraph/index';
import Heading6 from '../../Typography/H6/index';

const MessageItem = props => (
  <div className={`${styles.container}`}>
    <div className={`${styles.avatar}`}>
      <Avatar imageURL={props.avatarImageURL} />
    </div>
    <div className={`${styles.details}`}>
      <Heading6 labelText={props.recipientName}/>
      <Paragraph label={props.lastMessage} size='small'/>
    </div>
  </div>
);

MessageItem.defaultProps = {};

MessageItem.propTypes = {
  /** The location of the image for avatar */
  avatarImageURL: PropTypes.string.isRequired,
  /** The name of the message recipient/group */
  recipientName: PropTypes.string.isRequired,
  /** The last message that the recipient received */
  lastMessage: PropTypes.string.isRequired
};

export default MessageItem;
