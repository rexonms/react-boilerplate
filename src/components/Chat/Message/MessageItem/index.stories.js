import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { withKnobs } from '@storybook/addon-knobs/react';
import { withInfo } from '@storybook/addon-info';

import MessageItem from './index';

const stories = storiesOf('components/Chat/Message/MessageItem', module);
export const props = {
  avatarImageURL: 'http://dummyimage.com/mediumrectangle/222222/eeeeee',
  recipientName: 'Mr. Smith',
  lastMessage: 'Howdy!',
  email: 'mr.smith@gmail.com',
  imageAltText: 'Mr. Smith',
}

stories.addDecorator(withKnobs);
stories.addDecorator(story => (story()));
stories.add('MessageItem', withInfo(`
  A Stateful Component example that toggles the bulb state based on CHAT_BOX_USER click event.
  ~~~js
    <MessageItem
      avatarImageURL: 'http://dummyimage.com/mediumrectangle/222222/eeeeee',
      recipientName: 'Mr. Smith',
      lastMessage: 'Howdy!',
      email: 'mr.smith@gmail.com',
        imageAltText: 'Mr. Smith',
    />
  ~~~
`)(() => (
  <MessageItem
    {...props}
    onClick={action(this.props.email)}
  />
)));
