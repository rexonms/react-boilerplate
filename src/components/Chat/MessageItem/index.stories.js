import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, select } from '@storybook/addon-knobs/react';
import { withInfo } from '@storybook/addon-info';

import MessageItem from './index';

const stories = storiesOf('Components/Chat/MessageItem', module);
stories.addDecorator(withKnobs);

// Knobs as dynamic variables.
stories.addDecorator(story => (story()));

stories.add('Standard', withInfo(`
  Message items are the blocks that is displayed in message list. 
  ~~~js
    <MessageItem
      avatarImageURL="avatarImageURL"
      recipientName="recipientName"
      lastMessage="lastMessage"
    />  ~~~
`)(() => {
  return (
    <MessageItem
      avatarImageURL="http://dummyimage.com/mediumrectangle/222222/eeeeee"
      recipientName="John Smith"
      lastMessage="Howdy! my little friend. This has been "
    />
  )
}));

