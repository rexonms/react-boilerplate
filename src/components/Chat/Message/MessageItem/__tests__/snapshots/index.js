import React from 'react';
import { shallow } from 'enzyme';

import MessageItem from '../../index';
import { props } from './../../index.stories';

// https://hackernoon.com/oh-snap-snapshots-unit-integration-testing-with-jest-enzyme-13cc18aecb7b
it('Renders component correctly', () => {
  const component = shallow(<MessageItem {...props} onClick={() => {}} />);
  expect(component).toMatchSnapshot();
});
