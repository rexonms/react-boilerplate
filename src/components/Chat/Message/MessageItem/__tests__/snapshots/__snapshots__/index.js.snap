// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders component correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MessageItem
    avatarImageURL="http://dummyimage.com/mediumrectangle/222222/eeeeee"
    email="mr.smith@gmail.com"
    imageAltText="Mr. Smith"
    lastMessage="Howdy!"
    onClick={[Function]}
    recipientName="Mr. Smith"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="inner"
>
          <div
                    className="avatar"
          >
                    <Avatar
                              imageAltText="Mr. Smith"
                              imageURL="http://dummyimage.com/mediumrectangle/222222/eeeeee"
                    />
          </div>
          <div
                    className="details"
          >
                    <Heading6
                              labelText="Mr. Smith"
                    />
                    <Paragraph
                              color=""
                              label="Howdy!"
                              size="small"
                    />
          </div>
</div>,
        <button
          onClick={[Function]}
/>,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="avatar"
>
              <Avatar
                            imageAltText="Mr. Smith"
                            imageURL="http://dummyimage.com/mediumrectangle/222222/eeeeee"
              />
</div>,
            <div
              className="details"
>
              <Heading6
                            labelText="Mr. Smith"
              />
              <Paragraph
                            color=""
                            label="Howdy!"
                            size="small"
              />
</div>,
          ],
          "className": "inner",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Avatar
                imageAltText="Mr. Smith"
                imageURL="http://dummyimage.com/mediumrectangle/222222/eeeeee"
/>,
              "className": "avatar",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "imageAltText": "Mr. Smith",
                "imageURL": "http://dummyimage.com/mediumrectangle/222222/eeeeee",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Heading6
                  labelText="Mr. Smith"
/>,
                <Paragraph
                  color=""
                  label="Howdy!"
                  size="small"
/>,
              ],
              "className": "details",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "labelText": "Mr. Smith",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "color": "",
                  "label": "Howdy!",
                  "size": "small",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="inner"
>
            <div
                        className="avatar"
            >
                        <Avatar
                                    imageAltText="Mr. Smith"
                                    imageURL="http://dummyimage.com/mediumrectangle/222222/eeeeee"
                        />
            </div>
            <div
                        className="details"
            >
                        <Heading6
                                    labelText="Mr. Smith"
                        />
                        <Paragraph
                                    color=""
                                    label="Howdy!"
                                    size="small"
                        />
            </div>
</div>,
          <button
            onClick={[Function]}
/>,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="avatar"
>
                <Avatar
                                imageAltText="Mr. Smith"
                                imageURL="http://dummyimage.com/mediumrectangle/222222/eeeeee"
                />
</div>,
              <div
                className="details"
>
                <Heading6
                                labelText="Mr. Smith"
                />
                <Paragraph
                                color=""
                                label="Howdy!"
                                size="small"
                />
</div>,
            ],
            "className": "inner",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Avatar
                  imageAltText="Mr. Smith"
                  imageURL="http://dummyimage.com/mediumrectangle/222222/eeeeee"
/>,
                "className": "avatar",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "imageAltText": "Mr. Smith",
                  "imageURL": "http://dummyimage.com/mediumrectangle/222222/eeeeee",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Heading6
                    labelText="Mr. Smith"
/>,
                  <Paragraph
                    color=""
                    label="Howdy!"
                    size="small"
/>,
                ],
                "className": "details",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "labelText": "Mr. Smith",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "color": "",
                    "label": "Howdy!",
                    "size": "small",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
